docs/docs/advanced/docker.mdx

```diff
@@ -0,0 +1,221 @@
+
+All core functionalities (e.g. editors, compilers, formatters, code execution, etc) run in the browser.
+However, some features require [external services](./services.mdx) which depend on server-side implementations (e.g. [sharing](../features/share.mdx) short URLs, [broadcast server](../features/broadcast.mdx), etc).
+The docker setup described here provides out-of-the-box implementations for self-hosting these services.
+See below for the list of provided [services](#services).
+
+This allows self-hosted instances to have the same features as the [hosted app](https://livecodes.io).
+
+:::warning Note
+Most self-hosted instances will not require this setup. The static app should work just fine using other simpler [self-hosting methods](../features/self-hosting.mdx#guide).
+Only use the docker setup if you need to self-host [these services](#services).
+:::
+
+## Example
+
+This is an example of a self-hosted instance deployed to a VPS using the included Docker setup:
+
+https://vps.livecodes.io/
+
+Setting it up can be as simple as running the following command:
+
+```sh
+HOST_NAME=vps.livecodes.io docker compose up --build -d
+```
+
+## Requirements
+
+- Docker
+- Docker Compose
+- Git (optional - for pulling updates)
+
+<details>
+<summary>
+Install Git, Docker and Docker Compose on Ubuntu
+</summary>
+
+This script installs git, docker, and docker compose on Ubuntu:
+
+```sh
+#!/bin/bash
+
+# Update package lists
+sudo apt update
+
+# Install Git
+sudo apt install -y git
+
+# Install Docker
+sudo apt install -y ca-certificates curl gnupg lsb-release
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt update
+sudo apt install -y docker-ce docker-ce-cli containerd.io
+
+# Add current user to docker group
+sudo usermod -aG docker $USER
+newgrp docker # Apply group changes immediately
+
+# Install Docker Compose
+sudo apt install -y docker-compose
+```
+
+To use this script:
+
+**Save**
+
+Save the script to a file, for example: `install_docker_ubuntu.sh`.
+
+**Make Executable**
+
+Make the script executable with:
+
+```sh
+chmod +x install_docker_ubuntu.sh
+```
+
+**Run**
+
+Execute the script using:
+```sh
+sudo ./install_docker_ubuntu.sh
+```
+
+It's recommended to run it with `sudo` to ensure all necessary permissions are granted.
+
+**Verify**
+
+Verify the installations with:
+
+```sh
+git --version
+docker --version
+docker compose version
+```
+</details>
+
+## Getting Started
+
+Clone the repo:
+
+```sh
+git clone https://github.com/live-codes/livecodes.git
```

Comment:
I see the repo already has releases, wouldn't it be better to rely on downloading the latest release and extracting it?
1. Removes the need for installing `git`
2. Ensures the users would be getting a stable release, in case `main` got broken at any point

Reply:
I'll keep the answers below, so that we have a continuous thread.